src/object.di

```diff
@@ -595,6 +595,59 @@ inout(V) get(K, V)(inout(V[K])* aa, K key, lazy inout(V) defaultValue)
     return (*aa).get(key, defaultValue);
 }
 
+private void _destructRecurse(S)(ref S s)
+    if (is(S == struct))
+{
+    import core.internal.traits : hasElaborateDestructor;
+
+    static if (__traits(hasMember, S, "__dtor"))
+        s.__dtor();
+
+    foreach_reverse (i, ref field; s.tupleof)
+    {
+        static if (hasElaborateDestructor!(typeof(field)))
+            _destructRecurse(field);
+    }
+}
+
+private void _destructRecurse(E, size_t n)(ref E[n] arr)
+{
+    import core.internal.traits : hasElaborateDestructor;
+
+    static if (hasElaborateDestructor!E)
+    {
+        foreach_reverse (ref elem; arr)
+            _destructRecurse(elem);
+    }
+}
+
+// Public and explicitly undocumented
+void _postblitRecurse(S)(ref S s)
+    if (is(S == struct))
+{
+    import core.internal.traits : hasElaborateCopyConstructor;
+
+    foreach (ref field; s.tupleof)
+    {
+        static if (hasElaborateCopyConstructor!(typeof(field)))
+            _postblitRecurse(field);
+    }
+    static if (__traits(hasMember, S, "__postblit"))
+        s.__postblit();
+}
+
+// Ditto
+void _postblitRecurse(E, size_t n)(ref E[n] arr)
```

Comment:
Maybe this should forward to a `posblitArrayRecurse(T)(T[] arr)` function that does not depend on `N`?

The current solution here will generate a new function for every different array size for a given T, which is less than ideal.

On the other hand, I'll assume the compiler can exploit compile time info and unroll the code for small arrays, generating more efficient code?

Maybe something like
```
if (N > 4)
{
    // Forward to generic array implementation
    _postblitArrayRecurse(arr);
}
else
{
    // Do this here so compiler can exploit N's value
    ...
}
```
Or maybe this is premature optimisation.